Client/mods/deathmatch/logic/luadefs/CLuaColShapeDefs.cpp

```diff
@@ -235,11 +248,18 @@ int CLuaColShapeDefs::CreateColPolygon ( lua_State* luaVM )
                 CClientColPolygon* pShape = CStaticFunctionDefinitions::CreateColPolygon ( *pResource, vecPosition );
                 if ( pShape )
                 {
-                    // Get the points
-                    while ( argStream.NextCouldBeNumber () && argStream.NextCouldBeNumber ( 1 ) )
+                    // Get additional points
+                    while ( argStream.NextIsVector2D () )
+                    {
+                        CVector2D vecPoint;
+                        argStream.ReadVector2D ( vecPoint );
+                        vecPointList.push_back ( vecPoint );
```

Comment:
Why does `vecPointList` need to exist if its sole purpose is to perform `pShape->AddPoint( vecPointList[i] );`?

Can't this line be replaced with `pShape->AddPoint( vecPoint );`? 

(This is an honest question. There's probably a very good reason as to why you're doing this!)

Reply:
It is needed to store first 4 required vectors (position + 3 points),. An alternative would be to declare 4 separate variables. And since the list already exists and must be iterated over anyway, I thought it's reasonable to reuse it by pushing optional points to it too. This way `AddPoint` is called only in one loop instead of two which is more clean IMHO :) I don't know much about performance aspects here though.